zproject/computed_settings.py

```diff
@@ -312,7 +312,13 @@ def get_dirs(self) -> List[PosixPath]:
         DATABASES["default"]["OPTIONS"]["sslmode"] = REMOTE_POSTGRES_SSLMODE
     else:
         DATABASES["default"]["OPTIONS"]["sslmode"] = "verify-full"
-
+elif get_config("postgresql", "database_user") != "zulip":
+    if get_secret("postgres_password") is not None:
+        POSTGRESQL_DATABASE_PASSWORD = get_secret("postgres_password")
+        DATABASES["default"].update(
+            PASSWORD=get_secret("postgres_password"),
+            HOST="localhost",
+        )
```

Comment:
Hmm, I don't like the duplication here; we probably want to do a broader refactoring for this section.

Reply:
@timabbott This seemed the most logical way to do it, considering this is how it is done for instances where the database is stored remotely and for the development environment.

The reason line 317 also places the password in its own setting is due to `scripts/setup/postgresql-init-db` needing to be able to grab the password. I can't seem to think of a better way of doing this one.

Reply:
One way to fix this would be to make `get-django-setting` more general:
```diff
diff --git scripts/get-django-setting scripts/get-django-setting
index 9557b3abad..1f5adf22a1 100755
--- scripts/get-django-setting
+++ scripts/get-django-setting
@@ -1,14 +1,19 @@
 #!/usr/bin/env python3
 import os
 import sys
+from operator import getitem
+from functools import reduce

 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 sys.path.append(BASE_DIR)
 from scripts.lib.setup_path import setup_path

 setup_path()

 os.environ["DJANGO_SETTINGS_MODULE"] = "zproject.settings"
 from django.conf import settings

-print(getattr(settings, sys.argv[1]))
+found_setting = getattr(settings, sys.argv[1])
+found_setting = reduce(getitem, sys.argv[2:], found_setting)
+
+print(found_setting)
```

...which would let us call `./scripts/get-django-setting DATABASES default PASSWORD`.